### ISSUE TITLE
[Request] Vector Primitive Shape Panel

### ISSUE THREAD

**Issue by Jose-Moreno** [pencil2d/pencil#571]
This feature request is made by @Piehigh 

> I have been playing with pencil 2d abit and i think that it is important to have a shapes tool like in paint or in ms office products. This would make drawing so much easier and would be a great help. Also I think that it would also be nice to have layers. This would mean i could have different body parts as a different layer so i could keep the body consistent while still being able to change the arms without changing the body.

![pencil2d shape panel](https://cloud.githubusercontent.com/assets/4912907/20270048/97387e30-aa53-11e6-9cbe-f90a0f949f00.png)

**Comment by Jose-Moreno** [comment]
I agree on having more vector tools to allow for better drawing / inking of your hand drawn rough sketches, however I'd think we have to be careful as to how many tools to include. I mean I'd rather have a freehand / bezier tool to ink in the vector layer than to have multiple vector primitives or pre-baked shapes. This go tied in with the dysfunctional poly-line tool discussion.

Also when he speaks about having layers I think he means about having something akin to a swtich layer in anime studio / moho where you can have multiple layers nested inside a single switch layer (which is like a group layer) and it only allows for visibility of a single nested layer one at a time allowing the animator to "switch" between one or the other while animating.

If that's it, I think Synfig (www.synfig.com) can do this already, so we should discuss if we actually need this secondary thought in Pencil2D.

**Comment by Piehigh** [comment]
Thank you for fixing my mistake on where I posted this. I'm new to github and still learning to program which is why I am very interested at the source code even though I haven't started editing it and playing with it my self.  Also you are right in what I meant by separate layers (my apologies for not making it clear).

**Comment by Jose-Moreno** [comment]
@Piehigh Don't worry! I had a lot of issues when I first started using Github. Hopefully in the future you can aim to help the project. It sure needs all the help that it can get :smile:  For now, requests are getting queued meanwhile developers stabilize the software, but if they are open someone will work on it eventually! Thanks a lot! :smile:
